chain/client/src/view_client.rs

```diff
@@ -246,6 +250,62 @@ impl ViewClientActor {
         }
     }
 
+    /// Returns maintenance windows by account.
+    fn get_maintenance_windows(
+        &self,
+        account_id: AccountId,
+        genesis_config: GenesisConfig,
+    ) -> Result<MaintenanceWindowsView, near_chain::Error> {
+        let epoch_manager = EpochManager::new_from_genesis_config(
+            self.runtime_adapter.store().clone(),
+            &genesis_config,
+        )?;
+        let chain_store = ChainStore::new(
+            self.runtime_adapter.store().clone(),
+            genesis_config.genesis_height,
+            false,
+        );
+
+        let head = chain_store.head()?;
+        let epoch_id = epoch_manager.get_epoch_id(&head.last_block_hash)?;
+        let epoch_info: Arc<EpochInfo> = epoch_manager.get_epoch_info(&epoch_id)?;
+        let num_shards = self.runtime_adapter.num_shards(&epoch_id)?;
+        let cur_block_info = epoch_manager.get_block_info(&head.last_block_hash)?;
+        let next_epoch_start_height = epoch_manager
+            .get_epoch_start_height(cur_block_info.hash())?
+            + epoch_manager.get_epoch_config(&epoch_id)?.epoch_length;
+
+        let mut windows: MaintenanceWindowsView = Vec::new();
+        let mut start_block_of_window: Option<BlockHeight> = None;
+        let last_block_of_epoch = next_epoch_start_height - 1;
+
+        for block_height in head.height..next_epoch_start_height {
+            let bp = epoch_info.sample_block_producer(block_height);
+            let bp = epoch_info.get_validator(bp).account_id().clone();
+            let cps: Vec<AccountId> = (0..num_shards)
```

Comment:
please add some basic tests for this function. (let me know if you need help with that).

Reply:
Yes, could you give me some clues to test it?  There is no epoch in `start_all` in `jsonrpc-tests`.  Is there a better place to do the test on it?